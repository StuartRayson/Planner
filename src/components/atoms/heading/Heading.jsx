import classNames from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import './heading.scss';

const Heading = ({ type = 'h1', text, className }) => {
  const TagName = type;
  const classes = classNames(
    'a-heading',
    `a-heading--${type}`,
    className
  );

  return (
    <TagName className={classes}>
      {text}
    </TagName>
  );
};

Heading.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string.isRequired,
  type: PropTypes.string
};

export default Heading;
