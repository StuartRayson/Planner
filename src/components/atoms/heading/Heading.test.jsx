import React from 'react';
import { shallow } from 'enzyme';

import Heading from './Heading.jsx';

describe('<Heading />', () => {
  it('renders without crashing', () => {
    shallow(<Heading text="title" />);
  });

  it('should render h1 tag', () => {
    const heading = shallow(<Heading text="title" />);
    expect(heading.find('h1').length).toEqual(1);
  });

  it('should render h4 tag', () => {
    const heading = shallow(<Heading type="h4" text="title" />);
    expect(heading.find('h4').length).toEqual(1);
  });

  it('should render inner text', () => {
    const heading = shallow(<Heading type="h1" text="title" />);
    expect(heading.text()).toEqual('title');
  });

  it('should render and match snapshot with basic config', () => {
    const wrapper = shallow(
      <Heading type="h1" text="title" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
