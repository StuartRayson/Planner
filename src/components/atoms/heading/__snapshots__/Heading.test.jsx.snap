// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render and match snapshot with basic config 1`] = `
<h1
  className="a-heading a-heading--h1"
>
  title
</h1>
`;
